### ISSUE TITLE
IPAM creating range of size 1 for no good reason

### ISSUE THREAD

**Issue by bboreham** [weaveworks/weave#2009]
Note the `Size: 1` below, from https://github.com/weaveworks/weave/issues/1996#issuecomment-188381865

```
host1:~$ weave launch-router --init-peer-count=3
host2:~$ weave launch-router --init-peer-count=3 host1
host3:~$ weave launch-router --init-peer-count=3 host1 host2
host1:~$ weave run -ti alpine /bin/sh
host2:~$ weave run -ti alpine /bin/sh
host3:~$ weave run -ti alpine /bin/sh
host3:~$ weave report -f '{{json .IPAM.Entries}}' | jq .
[
  {
    "Version": 1,
    "IsKnownPeer": true,
    "Nickname": "host3",
    "Peer": "d2:e2:3d:cc:92:05",
    "Size": 524288,
    "Token": "10.32.0.0"
  },
  {
    "Version": 3,
    "IsKnownPeer": true,
    "Nickname": "host1",
    "Peer": "d6:2e:5b:f6:71:fa",
    "Size": 262144,
    "Token": "10.40.0.0"
  },
  {
    "Version": 0,
    "IsKnownPeer": true,
    "Nickname": "host2",
    "Peer": "8e:7c:37:fb:89:57",
    "Size": 262143,
    "Token": "10.44.0.0"
  },
  {
    "Version": 0,
    "IsKnownPeer": true,
    "Nickname": "host1",
    "Peer": "d6:2e:5b:f6:71:fa",
    "Size": 1,
    "Token": "10.47.255.255"
  }
]
```
(NB: it may take a while for this peer to get an up-to-date view, so if the above does not contain entries for all peers, then repeat until it does)

**Comment by brb** [comment]
The problem is that `host2` asks `host1` to donate some space within the `10.32.0.1-10.47.255.254` range (broadcast addresses are excluded [here]( https://github.com/weaveworks/weave/blob/v1.4.5/ipam/http.go#L34)). `host1` splits its owned range into two halves and it donates the second half excluding the `10.47.255.255` broadcast address (because it does not belong to the asked range).

I suggest to fix it by including the broadcast addresses when asking for a donation.

**Comment by bboreham** [comment]
SGTM

**Comment by bboreham** [comment]
I think I might have fixed this in #2012 by accident.

**Comment by brb** [comment]
Yep, it fixed the issue. Thanks.

I've added a test for this issue (https://github.com/weaveworks/weave/commit/02c1e386a4fd55888bc18e1e52120bdd18dd5cd4), so feel free to cherry-pick it into your branch (some IP alignment in expectations might be needed).

**Comment by awh** [comment]
Closed by #2012.
